src/opm/io/eclipse/ESmry.cpp

```diff
@@ -55,46 +57,34 @@ namespace Opm { namespace EclIO {
 
 ESmry::ESmry(const std::string &filename, bool loadBaseRunData)
 {
-    std::string rootN;
-    bool formatted=false;
+
+    boost::filesystem::path inputFileName(filename);
+    boost::filesystem::path rootName = inputFileName.parent_path() / inputFileName.stem();
     
-    char buff[PATH_MAX];
+    std::vector<bool> formattedVect;
     
-    if (getcwd( buff, PATH_MAX )==NULL){
-        throw std::invalid_argument("failed when trying to get current working directory");
-    }
-
-    std::string currentWorkingDir(buff);
-
-    std::string smspec_filen;
-
-    if (filename.substr(filename.length() - 7, 7) == ".SMSPEC") {
-        rootN = filename.substr(0,filename.length() -7);
-    } else if (filename.substr(filename.length() -8, 8) == ".FSMSPEC") {
-        rootN=filename.substr(0,filename.length() -8);
-        formatted = true;
-    } else {
-        rootN = filename;
+    if ((inputFileName.extension()!=".SMSPEC") && (inputFileName.extension()!=".FSMSPEC")){
+        throw std::invalid_argument("Inptut file should have extension .SMSPEC or .FSMSPEC");
     }
 
-    path = currentWorkingDir;
-    updatePathAndRootName(path, rootN);
+    bool formatted = inputFileName.extension()==".SMSPEC" ? false : true;
+    formattedVect.push_back(formatted);
+    
+    boost::filesystem::path path = boost::filesystem::current_path();;
 
-    if (formatted) {
-        smspec_filen = path + "/" + rootN + ".FSMSPEC";
-    } else {
-        smspec_filen = path + "/" + rootN + ".SMSPEC";
-    }
+    updatePathAndRootName(path, rootName);
+    
+    boost::filesystem::path smspec_file = path / rootName += inputFileName.extension();
```

Comment:
Assignment and `+=` in the same statement might be well defined, but it looks weird to me. I guess this could just be:
```
... smspec_file = path / rootname + inputFileName.extension();
```

Reply:
Agree, what you are proposing looks more like what I was expecting. But this result in compiler error. 
```
error: no match for operator+ (operand types are boost::filesystem::path and boost::filesystem::path)
```
Here is the reference I have used. https://www.boost.org/doc/libs/1_69_0/libs/filesystem/doc/reference.html#path-concatenation

Reply:
Depending on what you're actually trying to express, I might write this in one of two ways

1.  ```C++
    rootName += inputFileName.extension();
    boost::filesystem::path smspec_file = path / rootName;
    ```

2.
     ```C++
    boost::filesystem::path smspec_file = path / (rootName + inputFileName.extension());
    ```

Reply:
The first is not an option since this will change content of variable rootName. I'm therefor using number 2. I have made similar change 3 other places in the code for consistency.